### PR TITLE
Add MD5 Hashing to Test Framework

### PR DESCRIPTION
**Pull request by mxwli**
This is to allow for the incorporation of a CypherLogicTest into the framework.

### CODE REVIEW

File: test/include/test_runner/test_group.h [hunk 1]
```diff
@@ -29,6 +29,9 @@ struct TestStatement {
     std::string batchStatmentsCSVFile;
     std::optional<std::string> connName;
     bool reloadDBFlag = false;
+    bool expectHash = false;
+    uint64_t expectedNumValuesHashed;
```
Comment (ray6080) [inline test/include/test_runner/test_group.h:33]: Double check if you can reuse `expectedNumTuples`.

File: test/include/test_runner/test_group.h [hunk 2]
```diff
@@ -29,6 +29,9 @@ struct TestStatement {
...
+    std::string hashValueExpected;
```
Comment (ray6080) [inline test/include/test_runner/test_group.h:34]: `hashValueExpected` -> `expectedHashValue`.

File: src/include/common/md5.h [hunk 3]
```diff
@@ -0,0 +1,98 @@
+#pragma once
+/*
+** This code taken from the SQLite test library (can be found at
+** https://www.sqlite.org/sqllogictest/doc/trunk/about.wiki).
+** Originally found on the internet. The original header comment follows this comment.
+** The code has been heavily modified and reformatted
```
Comment (ray6080) [inline src/include/common/md5.h:6]: Better to say `The code has been refactored, but the algorithm stays the same.`?

File: test/test_files/md5testing/all_shortest_path_md5.test
Comment (ray6080) [inline test/test_files/md5testing/all_shortest_path_md5.test:1]: Instead of keeping a separate test file using md5, we should directly replace existing tests, which can reduce duplicated tests.

File: test/test_files/md5testing/order_by_md5.test
Comment (ray6080) [inline test/test_files/md5testing/order_by_md5.test:1]: Ditto. Let's directly replace existing tests.

File: test/test_runner/test_runner.cpp [hunk 4]
```diff
@@ -129,6 +150,19 @@ std::vector<std::string> TestRunner::convertResultToString(
     return actualOutput;
 }
 
+std::pair<std::string, uint64_t> TestRunner::convertResultToMD5Hash(QueryResult& queryResult) {
+    uint64_t numValuesHashed;
```
Comment (ray6080) [inline test/test_runner/test_runner.cpp:154]: Instead of comparing numValues, why not compare numTuples, since you are hashing on tuples? Comparing of num tuples looks simpler (for users) and consistent with what we are currently doing.

File: test/test_files/md5testing/md5.test
Comment (ray6080) [inline test/test_files/md5testing/md5.test:1]: Can you add a test case where we have null values in test?

**Review by ray6080 — APPROVED** [review]
Generally looks good! Thanks! 
Let's update the documentation on the testing framework.

Can you also open an issue that we can support `md5` utility function with what you have under `common`?

File: test/test_files/md5testing/order_by_md5.test
Comment (mxwli) [reply test/test_files/md5testing/order_by_md5.test:1]: Should we delete or comment out original expected outputs? I think that keeping the plain text there would still help with debugging if the hash test case fails.

Comment (ray6080) [reply test/test_files/md5testing/order_by_md5.test:1]: Agree. Can we keep them as comments?

File: test/test_runner/test_runner.cpp (hunk 4 above)
Comment (mxwli) [reply test/test_runner/test_runner.cpp:154]: Storing numTuples is probably the better option. It originally compared numValues because that's what SQLLogicTest does.

Comment (ray6080) [reply test/test_runner/test_runner.cpp:154]: I see. Do you think if we change it to comparing by numTuples, it will the transformation harder?

Comment (mxwli) [reply test/test_runner/test_runner.cpp:154]: It looks like in their tests, the number of values is equivalent to the number of tuples multiplied by the number of columns in the output, which shouldn't be hard to transform.

File: test/test_files/md5testing/md5.test
Comment (mxwli) [reply test/test_files/md5testing/md5.test:1]: Yep. I also want to note that since SQLLogicTest outputs null values as `"NULL"` instead of the empty string, null values will be hashed as `"NULL"` in the next commit.
